tools/filterwriter/range.go

```diff
@@ -9,11 +9,17 @@ type matchRange struct{ first, last int }
 
 // allRanges returns every indexes of instance of pattern in b, or nil if pattern is not present in b.
 func allRanges(b, pattern []byte) (ranges []matchRange) {
-	for i, idx := 0, bytes.Index(b, pattern); idx != -1; idx = bytes.Index(b[i:], pattern) {
-		ranges = append(ranges, matchRange{idx + i, idx + i + len(pattern)})
+	i := 0
+	for {
+		sub := b[i:len(b)]
```

Comment:
if len(b) == 0 then `b[i:len(b)]` will cause an IndexOutOfRange error

Reply:
b is never empty here

Reply:
ok